include/lensfun/lensfun.h.in

```diff
@@ -284,7 +294,13 @@ struct LF_EXPORT lfMount
     bool Check ();
 
 private:
-    std::vector<char*> MountCompat;
+    /** keeps Compat, MountCompatPtrs and MountCompat in-sync*/
+    void RebuildPointers();
```

Comment:
Maybe we should create another class with the C-API data and help methods so we separate the internal from the API data.

Also, it would be nice if you could write tests for the new and refactored implementations, just to be sure that we don't break anything now or later.

Reply:
I'm not sure about 1). It still has to be a part of `lfMount` class and has to keep in-sync some of the C-API visible members, like `char **Compat;` and right now I'm not sure that I can generalize this functionality across different classes.
If it can't be generalized then just an `std::vector` and `Rebulild` method is the best and simplest solution than multiple tightly coupled classes.

As about 2). I'll think what can be done and how it can be properly tested.

Reply:
@payano 
2) is done

Reply:
1. I was thinking of that we clearly distinguish data members and the C-API data members, if we have them in a separate class they won't "clutter" the Mount class. But I'm okay with both solutions.
2. Is it possible to create a lensfun project instead of using your account?

Reply:
You mean codecov? As far as I understand it is attached to some particular repository and to make it work smoothly you have to add codecov github app to the account and enable it for the repo, then a simple github action will do the rest.
I'm not sure that there is any other easy way to do it, so we have to wait for the maintainers.
(and I'm still waiting for the maintainers to approve CI pipeline for this PR...)

Reply:
As for the data members, I guess there was an idea to make the data member transparent as possible between C and C++ and they actively using that fact that right now all the data members are in the same place in C and C++ even if the structures are a different and C++ versions have much more data members. That works, but it looks a bit fragile and non-flexible and can't be changes without significant API changes.

A good example of C and C++ API co-existing is Vulkan, but it is C API wrapped with C++. Lensfun is the opposite case - it's a C++ API wrapped with C.